test/jdk/java/lang/System/i18nEnvArg.java

```diff
@@ -0,0 +1,151 @@
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.Map;
+import jdk.test.lib.process.ProcessTools;
+
+public class i18nEnvArg {
+    final static String EUC_JP_TEXT = "\u6F22\u5B57";
+
+    /*
+     * Checks OS is Linux and OS has ja_JP.eucjp locale or not.
+     * Sets EUC_JP's environment variable and argunments against ProcessBuilder
+     */
+    public static void main(String[] args) throws Exception {
+        ProcessBuilder pb;
+        if (args.length == 0) {
+            var cmds = List.of(
+                "--add-modules=" + System.getProperty("test.modules"),
+                "-classpath",
+                System.getProperty("test.class.path"),
+                "-Dtest.jdk=" + System.getProperty("test.jdk"),
+                "-Dtest.class.path=" + System.getProperty("test.class.path"),
+                "-Dtest.modules=" + System.getProperty("test.modules"),
+                "i18nEnvArg",
+                "Start");
+            pb = ProcessTools.createTestJvm(cmds);
+            Map<String, String> environ = pb.environment();
+            environ.clear();
+            environ.put("LANG", "ja_JP.eucjp");
+        } else {
+            String jnuEncoding = System.getProperty("sun.jnu.encoding");
+            Charset dcs = jnuEncoding != null
+                ? Charset.forName(jnuEncoding)
+                : Charset.defaultCharset();
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            if (!dcs.equals(cs)) {
+                return;
+            }
+            var cmds = List.of(
+                "--add-modules=" + System.getProperty("test.modules"),
+                "--add-opens=java.base/java.lang=ALL-UNNAMED",
+                "-classpath",
+                System.getProperty("test.class.path"),
+                "i18nEnvArg$Verify",
+                EUC_JP_TEXT);
+            pb = ProcessTools.createTestJvm(cmds);
+            Map<String, String> environ = pb.environment();
+            environ.clear();
+            environ.put("LANG", "ja_JP.eucjp");
+            environ.put(EUC_JP_TEXT, EUC_JP_TEXT);
+        }
+        ProcessTools.executeProcess(pb)
+            .outputTo(System.out)
+            .errorTo(System.err)
+            .shouldHaveExitValue(0);
+    }
+
+    public static class Verify {
+        private final static int maxSize = 4096;
+        /*
+         * Verify environment variable and argument are encoded by Linux's eucjp or not
+         */
+        public static void main(String[] args) throws Exception {
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            byte[] euc = (EUC_JP_TEXT + "=" + EUC_JP_TEXT).getBytes(cs);
+            byte[] eucjp = "LANG=ja_JP.eucjp".getBytes(cs);
+            String s = System.getenv(EUC_JP_TEXT);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            PrintStream ps = new PrintStream(baos);
```

Comment:
Use `shouldNotContain()` to find the error message.

Reply:
I was suggesting `try (ByteArrayOutputStream baos = ...) {` so that no need to clean them up, but I see you removed them. But I prefer not to use `shouldNotContain("ERROR: ")` but to check the return value as before.